crates/anvil/src/eth/api.rs

```diff
@@ -2698,3 +2594,57 @@ fn determine_base_gas_by_kind(request: &TransactionRequest) -> U256 {
         _ => MIN_CREATE_GAS,
     }
 }
+
+/// Keeps result of a call to revm EVM used for gas estimation
+enum GasEstimationCallResult {
+    Success(u64),
+    OutOfGas,
+    Revert(Option<Bytes>),
+    EvmError(InstructionResult),
+}
+
+/// Converts the result of a call to revm EVM into a [GasEstimationCallRes].
+impl TryFrom<Result<(InstructionResult, Option<Output>, u64, State)>> for GasEstimationCallResult {
```

Comment:
we can keep this,
I think we can clean this up with a helper type after the larger alloy pr is through